### PR TITLE
feat(filteReddit): do not filter own posts

### LINKED ISSUE #3018: filtereddit - Allow your own posts (enable by default)
https://www.reddit.com/r/Enhancement/comments/4lilx7/feature_request_never_filter_posts_that_are/

### PR DESCRIPTION
**Pull request by ssonal**
Added an option to the filteReddit module to prevent filtering of own posts unless explicitly disabled.

Closes #3018

### CODE REVIEW

File: lib/modules/filteReddit.js [hunk 1]
```diff
@@ -59,6 +59,11 @@ module.options = {
 		noconfig: process.env.BUILD_TARGET === 'safari' || process.env.BUILD_TARGET === 'edge',
 		description: 'Show a \'filter visited links\' tab at the top of each subreddit, to easily toggle whether to filter visited posts.',
 	},
+	exludeOwnPosts: {
```
Comment (andytuba) [inline lib/modules/filteReddit.js]: Exclude -- spelling

File: lib/modules/filteReddit.js [hunk 2]
```diff
@@ -827,6 +832,16 @@ function scanEntries(ele) {
 			const postFlair = thing.getPostFlairText();
 			const postLink = thing.getPostLink();
 
+			// Do not apply any other filters if posts belong to me
+			if (module.options.exludeOwnPosts.value) {
+				const postAuthor = thing.getAuthor().toLowerCase();
+				// No standard marker for my own posts so compare against the logged in user
+				const myName = loggedInUser();
+				if (myName && (myName.trim().toLowerCase() === postAuthor)) {
```
Comment (andytuba) [inline lib/modules/filteReddit.js]: I think trim is overkill in this case.

File: lib/modules/filteReddit.js [hunk 3]
```diff
@@ -827,6 +832,16 @@ function scanEntries(ele) {
 			const postFlair = thing.getPostFlairText();
 			const postLink = thing.getPostLink();
 
+			// Do not apply any other filters if posts belong to me
```
Comment (andytuba) [inline lib/modules/filteReddit.js]: How about moving this above `const postTitle`, so  the function can quit as early as possible?

**Comment by andytuba** [comment]
A few small fixes and this looks great, thanks for contributing!

File: lib/modules/filteReddit.js (hunk 1 above)
Comment (ssonal) [inline lib/modules/filteReddit.js]: Silly me!

File: lib/modules/filteReddit.js (hunk 3 above)
Comment (ssonal) [inline lib/modules/filteReddit.js]: Sounds good.

**Comment by erikdesjardins** [comment]
Looks good, thanks!

**Pull request merged**